src/lib.rs

```diff
@@ -36,6 +36,23 @@ pub trait KeyBox: Index {
     fn verify(&self, msg: &[u8], sgn: &Self::Signature, index: NodeIndex) -> bool;
 }
 
+pub trait MultiKeychain: KeyBox {
+    type PartialMultisignature: Debug + Clone + Encode + Decode;
+    fn get_signers(partial: &Self::PartialMultisignature) -> BoolNodeMap;
+    fn add_signature(
+        partial: &mut Self::PartialMultisignature,
+        index: NodeIndex,
+        signature: &Self::Signature,
+    );
+    fn is_complete(partial: &Self::PartialMultisignature) -> bool;
+    fn verify_partial(
+        &self,
+        msg: &[u8],
+        partial: &Self::PartialMultisignature,
+        indices: &BoolNodeMap,
+    ) -> bool;
```

Comment:
Only a comment, not a change request: it might turn out that the interface with verify_partial is not ideal. In all schemes I'm aware of, one verifies each individual signature of a multisignature separately. So something like
```
verify_share(&self, msg: &[u8], s: Self::Signature, index:usize)
```
might be better. Anyway, this is easy to change later.

Reply:
Looking at the implementation of `PartiallyMultisigned::add_signature` line 146, I believe that the author meant `verify` that is inherited by `MultiKeychain` from `KeyBox` to be used as `verify_share` that you propose. And only check `is_complete` on the final multisignature.

Reply:
Yes, that was my intention. the MultiKeychain builds on top of KeyBox and adds signature aggrgation functionality.